apps/cli/src/commands/boot-challenger.ts

```diff
@@ -114,26 +116,82 @@ const onAssertionConfirmedCb = async (nodeNum: any, commandInstance: Vorpal.Comm
         }
     }
 
-    const assertionNode = await getAssertion(nodeNum);
-    commandInstance.log(`[${new Date().toISOString()}] Assertion data retrieved. Starting the submission process...`);
-    try {
-        await submitAssertionToReferee(
-            cachedSecretKey,
-            nodeNum,
-            assertionNode,
-            cachedSigner!.signer,
-        );
-        commandInstance.log(`[${new Date().toISOString()}] Submitted assertion: ${nodeNum}`);
-        lastAssertionTime = Date.now();
-    } catch (error) {
-        if (error && (error as Error).message && (error as Error).message.includes('execution reverted: "9"')) {
-            commandInstance.log(`[${new Date().toISOString()}] Could not submit challenge because it was already submitted`);
-            lastAssertionTime = Date.now();
-            return;
+    // Calculate the minimum time to submit an assertion
+    lastAssertionTime = lastAssertionTime || 0;
```

Comment:
Having `|| 0` cannot be right. If the `lastAssertionTime` is not set we have to pull it from onChain

Reply:
`lastAssertionTime` is currently used only for the health check notification system. I don't think its a good idea to add critical functionality to this variable. Also consider this will be set to `Date.now()` on boot.

Reply:
This has been removed and a core function added to retrieve the last assertionId and timestamp from chain.